openedx/core/djangoapps/user_api/api.py

```diff
@@ -0,0 +1,819 @@
+import copy
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse
+from django.utils.translation import ugettext as _
+from django_countries import countries
+
+import accounts
+import third_party_auth
+from edxmako.shortcuts import marketing_link
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.helpers import FormDescription
+from openedx.features.enterprise_support.api import enterprise_customer_for_request
+from student.forms import get_registration_extension_form
+from student.models import UserProfile
+
+
+def get_password_reset_response():
+    """Return a description of the password reset form.
+
+    This decouples clients from the API definition:
+    if the API decides to modify the form, clients won't need
+    to be updated.
+
+    See `user_api.helpers.FormDescription` for examples
+    of the JSON-encoded form description.
+
+    Returns:
+        HttpResponse
+
+    """
+    form_desc = FormDescription("post", reverse("password_change_request"))
+
+    # Translators: This label appears above a field on the password reset
+    # form meant to hold the user's email address.
+    email_label = _(u"Email")
+
+    # Translators: This example email address is used as a placeholder in
+    # a field on the password reset form meant to hold the user's email address.
+    email_placeholder = _(u"username@domain.com")
+
+    # Translators: These instructions appear on the password reset form,
+    # immediately below a field meant to hold the user's email address.
+    email_instructions = _(u"The email address you used to register with {platform_name}").format(
+        platform_name=configuration_helpers.get_value('PLATFORM_NAME', settings.PLATFORM_NAME)
+    )
+
+    form_desc.add_field(
+        "email",
+        field_type="email",
+        label=email_label,
+        placeholder=email_placeholder,
+        instructions=email_instructions,
+        restrictions={
+            "min_length": accounts.EMAIL_MIN_LENGTH,
+            "max_length": accounts.EMAIL_MAX_LENGTH,
+        }
+    )
+
+    return HttpResponse(form_desc.to_json(), content_type="application/json")
```

Comment:
Instead of returning an response, why don't we just return the form description `form_desc`? This way, we can make this function more focused. We can also just name this function to be `get_password_reset_form`